docs/SPEC.md

```diff
@@ -1426,6 +1427,176 @@ A query specification defines what data and operations to perform.
+
+When multiple modules both depend on a specific version of another module the maximum version of the minimum versions is used.
+Major versions of a module are considered different modules (they have different module paths), therefore multiple major versions of a module may be imported into the same Flux script.
+
+When multiple import declarations exist for the same module at most one import declaration must specify version information.
+
+Example
+
+```
+// a.flux
+package a
+
+import "foo@v1.1.0"
+```
+
+```
+// b.flux
+package b
+
+import "foo@v1.2.0"
+```
+
+```
+// main.flux
+@registry("modules", "http://localhost/modules")
+package main
+
+import "modules/a"
+import "modules/b"
+```
+
+Package `main` depends on module `foo` via both of the modules `a` and `b`.
+However `a` and `b` specify different versions of `foo`.
+The possible versions of `foo` include `1.1.0` and `1.2.0`.
+Flux will pick the maximum version of these possible versions, so version `1.2.0` of `foo` is used.
+This is sound because module `a` has specified that it needs at least version `1.1.0` of `foo` and that constraint is satisfied.
+
+### Registry API
+
+Modules can be published and downloaded over an HTTP API from a registry.
+Modules are immutable, once a version is published it cannot be modified, a new version must be published instead.
+
+The HTTP API will have the routes listed in the following table where `$base` is the anchor point of the API, `$module` is a module path without the registry name, and `$version` is a semantic version of the form `vMAJOR.MINOR.PATCH`.
+
+| Method | Path                          | Description                                                                                                                                                |
+| ------ | ----                          | -----------                                                                                                                                                |
+| GET    | $base/$module/@v/list         | Returns a list of known versions of the given module in plain text, one per line.                                                                          |
+| GET    | $base/$module/@v/$version.zip | Returns a zip file of the contents of the module at a specific version.                                                                                    |
+| GET    | $base/$module/@latest         | Returns the highest released version, or if no released versions exist the highest pre-release version of the given module in plain text on a single line. |
+| GET    | $base/$module/@pre            | Returns the highest pre-released version of the given module in plain text on a single line.                                                               |
+| POST   | $base/$module/@v/$version     | Publish a new version of the module where the POST body contains multipart formdata for the contents of the module.                                        |
```

Comment:
The `@v` delimiter is no longer necessary since a module path is at most two path elements, of the form `name/v#` where `#` is a number >= 2.  We needed a rule to distinguish module paths from package paths and it so happens we can reuse that rule for the registry API. 

Should we remove it?

Reply:
How does `POST` and `@pre` work---is there a way to specify that the version you are publishing is release or pre-release?